Libraries/LibWeb/DOM/Element.cpp

```diff
@@ -1874,6 +1874,54 @@ void Element::invalidate_style_after_attribute_change(FlyString const& attribute
     invalidate_style(StyleInvalidationReason::ElementAttributeChange);
 }
 
+bool Element::is_hidden() const
+{
+    if (layout_node() == nullptr)
+        return true;
+    if ((layout_node() != nullptr) && (layout_node()->computed_values().visibility() == CSS::Visibility::Hidden || layout_node()->computed_values().visibility() == CSS::Visibility::Collapse || layout_node()->computed_values().content_visibility() == CSS::ContentVisibility::Hidden))
+        return true;
+    for (ParentNode const* self_or_ancestor = this; self_or_ancestor; self_or_ancestor = self_or_ancestor->parent_or_shadow_host()) {
+        if (self_or_ancestor->is_element() && static_cast<DOM::Element const*>(self_or_ancestor)->aria_hidden() == "true")
+            return true;
+    }
+    return false;
+}
+
+bool Element::has_hidden_ancestor() const
+{
+    for (ParentNode const* self_or_ancestor = this; self_or_ancestor; self_or_ancestor = self_or_ancestor->parent_or_shadow_host()) {
+        if (self_or_ancestor->is_element() && static_cast<DOM::Element const*>(self_or_ancestor)->is_hidden())
+            return true;
+    }
+    return false;
+}
+
+bool Element::is_referenced() const
+{
+    bool is_referenced = false;
+    if (id().has_value()) {
```

Comment:
This can be an early return, and I don't think you need the `bool is_referenced` after that change.

Reply:
But I need to do the `aria_labelled_by_or_default().contains_slow(id().value()` check within the `for_each_in_inclusive_subtree_of_type` lambda.

So if a normal early return within that lambda won’t cause the `Element::is_referenced` function to return, is there some other way from inside that lambda I can cause an early return of the `Element::is_referenced` function?